workflow-templates/im-deploy-az-database.yml

```
@@ -56,25 +56,36 @@ on:
   #   types: [<deployable_name>_deploy] # TODO: Replace <deployable_name>.  This will be used in the 'Deploy Multiple Items' workflow to target this deployment workflow.
 
 env:
-  ENVIRONMENT: ${{ github.event.inputs.environment }}
-  RELEASE_TAG: ${{ github.event.inputs.tag }} # This is the tag that we'll be deploying
+  ENVIRONMENT: ${{ inputs.environment-or-target }}
+  RELEASE_TAG: ${{ inputs.tag == 0 && github.ref_name || inputs.tag }} # This is the tag that we'll be deploying
   DB_SERVER_PORT: '1433' # TODO: Verify that this is the correct port
   MIGRATIONS_FOLDER: '' # TODO: Add the path to the migrations folder
   MIGRATIONS_HISTORY_TABLE: 'MigrationHistory' # TODO: Verify this table name
   TIMEZONE: 'america/denver' # TODO: Verify timezone
 
 jobs:
-  verify-tag-exists:
+  set-vars:
+    needs: [examine-inputs]
     runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
 
+    outputs:
+      # To use these values: ${{ needs.set-vars.outputs.<OUTPUT_NAME> }}
+      TARGET_RESOURCE_GROUP: ${{ steps.set-variables.outputs.TARGET_RESOURCE_GROUP }}
+      PRIMARY_RESOURCE_GROUP: ${{ steps.set-variables.outputs.PRIMARY_RESOURCE_GROUP }}
+      APP_INSIGHTS_NAME: ${{ steps.set-variables.outputs.APP_INSIGHTS_NAME }}
+      APP_INSIGHTS_ANNOTATION: ${{ steps.set-variables.outputs.APP_INSIGHTS_ANNOTATION }}
+      GITHUB_SECRETS_ENVIRONMENT: ${{ steps.set-variables.outputs.GITHUB_SECRETS_ENVIRONMENT }}
+
     steps:
-      - run: |
+      - name: Annotate Inputs
+        run: |
           echo $'
-          | Build Arguments | Value                    |
-          | ---             |  ---                     |
-          | Environment     | `${{ env.ENVIRONMENT }}` |
-          | Tag             | `${{ env.RELEASE_TAG }}` |' >> $GITHUB_STEP_SUMMARY
-                
+          | Deploy Arguments          | Value                    |
+          | ---                       |  ---                     |
+          | Environment               | `${{ env.ENVIRONMENT }}` |
+          | Release                   | `${{ env.RELEASE_TAG  }}`  |
+          | Workflow Branch/Tag       | `${{ github.ref_name }}` - SHA: `${{ github.sha }}` |' >> $GITHUB_STEP_SUMMARY
```

I like this addition. It probably belongs in the other deployment templates that use `workflow_dispatch`.

Maybe change the label though:
```suggestion
          | Workflow Source           | `${{ github.ref_name }}` - SHA: `${{ github.sha }}` |' >> $GITHUB_STEP_SUMMARY
```